sql/sql_select.cc

```diff
@@ -946,9 +946,10 @@ bool Sql_cmd_select::check_privileges(THD *thd) {
 
 bool Sql_cmd_dml::check_all_table_privileges(THD *thd) {
   // Check for all possible DML privileges
+  TABLE_LIST *first_not_own_table = lex->first_not_own_table();
 
-  for (TABLE_LIST *tr = lex->query_tables; tr != nullptr;
-       tr = tr->next_global) {
+  for (TABLE_LIST *tr = lex->query_tables;
```

Comment:
Is only select affected, or there are others as well?

Reply:
This bug affect UPDATEs, INSERTs and DELETEs. It could hypothetically affect SELECTs, but as far as I know, it's not possible to create a trigger on SELECTs.

Although this code is in `sql_select.cc` file, it's the implementation of the `Sql_cmd_dml` class, so it's inherited by all DML commands. All descendant classes have own implementation of `<...>::check_privileges()`, but they all also call `Sql_cmd_dml::check_all_table_privileges()`, so changes here affect all DML commands.